File: services/api/app/models/virtual_machine.rb
# Copyright (C) The Arvados Authors. All rights reserved.
#
# SPDX-License-Identifier: AGPL-3.0

class VirtualMachine < ArvadosModel
  include HasUuid
  include KindAndEtag
  include CommonApiTemplate

  has_many(:login_permissions,
           -> { where("link_class = 'permission' and name = 'can_login'") },
           foreign_key: 'head_uuid',
           class_name: 'Link',
           primary_key: 'uuid')

  api_accessible :user, extend: :common do |t|
    t.add :hostname
  end

  protected

  def permission_to_create
    current_user and current_user.is_admin
  end
  def permission_to_update
    current_user and current_user.is_admin
  end
end

File: sdk/go/arvados/virtual_machine.go
// Copyright (C) The Arvados Authors. All rights reserved.
//
// SPDX-License-Identifier: Apache-2.0

package arvados

import "time"

// VirtualMachine is an arvados#virtualMachine resource.
type VirtualMachine struct {
	UUID               string     `json:"uuid"`
	OwnerUUID          string     `json:"owner_uuid"`
	Hostname           string     `json:"hostname"`
	CreatedAt          *time.Time `json:"created_at"`
	ModifiedAt         *time.Time `json:"modified_at"`
	ModifiedByUserUUID string     `json:"modified_by_user_uuid"`
}

// VirtualMachineList is an arvados#virtualMachineList resource.
type VirtualMachineList struct {
	Items          []VirtualMachine `json:"items"`
	ItemsAvailable int              `json:"items_available"`
	Offset         int              `json:"offset"`
	Limit          int              `json:"limit"`
}
